google/resource_google_project.go

```diff
@@ -492,10 +501,39 @@ func resourceGoogleProjectUpdate(d *schema.ResourceData, meta interface{}) error
 		d.SetPartial("labels")
 	}
 
-	// ignore app_engine changes, they don't work anyways.
+	// App Engine App has changed
+	if ok := d.HasChange("app_engine"); ok {
+		app, err := expandAppEngineApp(d)
+		if err != nil {
+			return err
+		}
+		// ignore if app is now not set; that should force new resource using customizediff
+		if app != nil {
+			if old, new := d.GetChange("app_engine.#"); (old == nil || old.(int) < 1) && new != nil && new.(int) > 0 {
+				err = createAppEngineApp(config, pid, app)
+				if err != nil {
+					return err
+				}
+			} else {
+				log.Printf("[DEBUG] Updating App Engine App")
+				op, err := config.clientAppEngine.Apps.Patch(pid, app).UpdateMask("authDomain,servingStatus,featureSettings.splitHealthChecks").Do()
+				if err != nil {
+					return fmt.Errorf("Error creating App Engine application: %s", err.Error())
+				}
+
+				// Wait for the operation to complete
+				waitErr := appEngineOperationWait(config.clientAppEngine, op, pid, "App Engine app to update")
+				if waitErr != nil {
+					return waitErr
+				}
+				log.Printf("[DEBUG] Updated App Engine App")
+			}
+		}
+	}
```

Comment:
May as well d.SetPartial here in case someone comes and adds a new HasChange after this

Reply:
👍